### DISCUSSION TITLE
Wrap codeblocks w/ element?

### DISCUSSION THREAD

**Discussion by AutomateAaron** [pngwn/MDsveX#504] [Q&A] [answered]
I'm trying to wrap the code blocks w/ an element so that I can style them in a particular way.  My thinking was to use the [highlighter](https://mdsvex.com/docs#highlight) configuration to pass in a custom function which wraps the default function.  However, I'm having trouble importing [the default function](https://github.com/pngwn/MDsveX/blob/095d10873c25c6650516a95d06e05be5494166a6/packages/mdsvex/src/transformers/index.ts#L555-L579).  Is there a way to import this?

**Comment by babakfp** [comment]
Hey
I don't know about the `code_highlight` function, but here is how I added code highlighting to my blog:

```js
import { mdsvex, escapeSvelte } from "mdsvex"
import { getHighlighter } from "shiki"

mdsvex({
	highlight: {
		highlighter: async (code, lang) => {
			const highlightedCode = await mdsvexHighlight(code, lang)
			return highlightedCode ? escapeSvelte(highlightedCode) : ""
		},
	},
}),

async function mdsvexHighlight(code, lang) {
	const highlighter = await getHighlighter({
		theme: "rose-pine",
		langs: ["html", "css", "js", "svelte", "php", "cmd"],
	})
	return highlighter.codeToHtml(code, { lang })
}
```

**Reply by AutomateAaron** [reply]
I appreciate it, but is it possible to import the default function?

**Reply by babakfp** [reply]
How did you import the `code_highlight` function? There was no export keyword in front of that function (look into the package folder inside the `node_modules` folder). You can copy the function source code and use it that way.

**Reply by AutomateAaron** [reply]
Would it be possible to export this function?  It'd be useful to keep my copied version updated as the project changes.

**Reply by babakfp** [reply]
I don't know. I guess you can ask the maintainer to export the function in the new release 🤷.

**Reply by AutomateAaron** [reply]
@pngwn would it be possible to export the `code_highlight` function?

**Reply by pngwn** [reply, marked as answer]
I think that's what they are asking. @AaronNBrock could you file an issue asking for this please? I'll try to take care of it soon.

**Reply by AutomateAaron** [reply]
Issue created: https://github.com/pngwn/MDsveX/issues/507
